packages/generator-common/src/compiler.ts

```diff
@@ -22,17 +22,31 @@ const logger = createLogger('compiler');
  * It recursively compiles all files ending with .ts
  * @param path - Directory to be compiled.
  * @param compilerOptions - Compiler options to be used
+ * @param includeExclude - Included and excluded files for compilation
  */
 export async function transpileDirectory(
   path: string,
-  compilerOptions: CompilerOptions
+  compilerOptions: CompilerOptions,
+  includeExclude: IncludeExclude = defaultIncludeExclude
 ): Promise<void> {
   logger.verbose(`Transpiling files in the directory: ${path} started.`);
 
-  const files = new GlobSync(resolve(path, '**/*.ts'), {
-    ignore: resolve(path, '**/*.d.ts')
-  }).found;
-  const program = await createProgram(files, compilerOptions);
+  const includes = includeExclude.include.map(include =>
+    resolve(path, include)
+  );
+  const excludes = includeExclude.exclude.map(exclude =>
+    resolve(path, exclude)
+  );
+  const allFiles: string[] = flatten(
+    includes.map(
+      include =>
+        new GlobSync(resolve(path, include), {
+          ignore: excludes
+        }).found
+    )
+  );
```

Comment:
I came across this way of searching for more than one pattern. This avoids having to do multiple searches I believe.
```
  const allFiles = new GlobSync(resolve(path, '**/*.ts'), {
    ignore: resolve(path, '{dist/**/*, **/*.d.ts, **/*.spec.ts, node_modules/**/*}')
  }).found;
```

Reply:
I checked and this works only for the excludes. For the includes it does not. So I would perhaps keep it the way it is, since the number of searches is not reduced.

Reply:
But joining with `|` seems to work for the include. I add a test to ensure it.

Reply:
For include, if its a single pattern, it works without giving any `{}`, so either like
`new GlobSync(resolve(path, '**/*.ts')` or like `new GlobSync(resolve(path, '{**/*.ts,}')` (not pretty). For more that one pattern in include this syntax worked, example: `new GlobSync(resolve(path, '{**/*.ts, **/*.d.ts}')`